crates/rerun_c/src/rerun.h

```diff
@@ -412,6 +412,23 @@ extern void rr_recording_stream_log(
     rr_recording_stream stream, rr_data_row data_row, bool inject_time, rr_error* error
 );
 
+// ----------------------------------------------------------------------------
+// Private functions
+
+/// PRIVATE FUNCTION: do not use.
+///
+/// Escape a single part of an entity path, returning an new null-terminated string.
+///
+/// The returned string must be freed with `_rr_free_string`.
+///
+/// Returns `nullptr` on failure (e.g. invalid UTF8, ore null bytes in the string).
+extern char* _rr_escape_entity_path_part(rr_string part);
+
+/// PRIVATE FUNCTION: do not use.
+///
+/// Must only be called with the results from `_rr_escape_entity_path_part`.
+extern void _rr_free_string(char* string);
```

Comment:
suggestion:
instead of doing the alloc/free dance a slightly safer alternative would be to use the "buffer in, count inout" pattern:

```
int rr_escape_entity_path_part(rr_string part, char* buffer, int buffer_size);
```

In this fairly common C pattern the desired count is returned but the buffer isn't fully written. Meaning one can either pass a null for buffer to just figure out the size one needs to allocate, or speculatively put in a buffer and check whether a second call is needed.

The advantage is that in many cases only one call is needed and that allocation stays with the caller.
The disadvantage is that when passing null buffer or too small buffer initially, this bottoms out in more work being done.

Reply:
don't quite understand why `new_entity_path` is not fully implemented in the c interface

Reply:
I went with this implementation because it was much faster to implement.

Reply:
> don't quite understand why `new_entity_path` is not fully implemented in the c interface

I don't understand the question. `new_entity_path` returns a `std::string`, so cannot be defined in C.

Reply:
you could pass in several `rr_string` and write to a single char* buffer